src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -741,7 +741,7 @@ int32_t NaClSysOpen(struct NaClAppThread  *natp,
   */
   
   fd_retval = NextFd(nap->cage_id);
-  if(hd) hd->lindfd = fd_retval;
+  if(hd) fd_retval = hd->lindfd = hd->d;
```

Comment:
We probably need a good bit of text explaining the lindfd thing now, also probably separate these.